src/connectors/youtube.js

```diff
@@ -174,25 +177,46 @@ function getVideoCategory(videoId) {
 	if (videoId === null) {
 		return null;
 	}
-	if (!categoryCache.has(videoId)) {
-		const url = `https://www.googleapis.com/youtube/v3/videos?part=snippet&id=${videoId}&key=${YT_API_KEY}`;
-		fetch(url).then((response) => {
+	if (categoryCache.has(videoId)) {
+		fetchCategoryId(videoId).then((category) => {
+			if (category === null) {
+				throw Error(`Failed to resolve category for ${videoId}`);
```

Comment:
Well, in this case the connector will fetch category for video id constantly. I suggest to use this algorithm:
```js
fetchCategoryId(videoId).then((category) => {
	if (category === null) {
		console.log(`Failed to resolve category for ${videoId}`);
	}
	categoryCache.set(videoId, category);
}).catch((err) => {
	console.log(err);
});
```
Add category in cache anyway, even if it's null.

Reply:
Should we tackle this seperately?